graspy/match/sgm.py

```diff
@@ -0,0 +1,335 @@
+
+        "rand" : some random point near :math:`J, (J+K)/2`, where K is some random doubly
+        stochastic matrix
+
+    max_iter : int, positive (default = 30)
+        Integer specifying the max number of FW iterations.
+        FAQ typically converges with modest number of iterations
+
+    shuffle_input : bool (default = True)
+        Gives users the option to shuffle the nodes of A matrix to avoid results
+        from inputs that were already matched
+
+    eps : float (default = 0.1)
+        A positive, threshold stopping criteria such that FW continues to iterate
+        while Frobenius norm of :math:`(P_{i}-P_{i+1}) > eps`
+
+
+    gmp : bool (default = True)
+        Gives users the option to the Quadratic Assignment (QAP)rather than the
+        Graph Matching Problem (GMP). This is accomplished through trivial
+        negation of the objective function.
+
+    Attributes
+    ----------
+
+    perm_inds_ : array, size (n,) where n is the number of vertices in the graphs fitted.
+        The indices of the optimal permutation (with the fixed seeds given) on the nodes of B,
+        found via SGM, to best minimize the objective function :math:`f(P) = trace(A^T PBP^T )`.
+
+
+    score_ : float
+        The objective function value of for the optimal permutation found.
+
+
+    References
+    ----------
+    .. [1] D. Fishkind, S. Adali, H. Patsolic, L. Meng, D. Singh, V. Lyzinski, C. Priebe,
+           Seeded graph matching, Pattern Recognit. 87 (2019) 203–215
+
+
+
+    """
+
+    def __init__(
+        self,
+        n_init=1,
+        init_method="barycenter",
+        max_iter=30,
+        shuffle_input=True,
+        eps=0.1,
+        gmp=True,
+    ):
+
+        if type(n_init) is int and n_init > 0:
+            self.n_init = n_init
+        else:
+            msg = '"n_init" must be a positive integer'
+            raise TypeError(msg)
+        if init_method == "rand":
+            self.init_method = "rand"
+        elif init_method == "barycenter":
+            self.init_method = "barycenter"
+            self.n_init = 1
+        else:
+            msg = 'Invalid "init_method" parameter string'
+            raise ValueError(msg)
+        if max_iter > 0 and type(max_iter) is int:
+            self.max_iter = max_iter
+        else:
+            msg = '"max_iter" must be a positive integer'
+            raise TypeError(msg)
+        if type(shuffle_input) is bool:
+            self.shuffle_input = shuffle_input
+        else:
+            msg = '"shuffle_input" must be a boolean'
+            raise TypeError(msg)
+        if eps > 0 and type(eps) is float:
+            self.eps = eps
+        else:
+            msg = '"eps" must be a positive float'
+            raise TypeError(msg)
+        if type(gmp) is bool:
+            self.gmp = gmp
+        else:
+            msg = '"gmp" must be a boolean'
+            raise TypeError(msg)
+
+    def fit(self, A, B, W1=[], W2=[]):
```

Comment:
I am okay with A and B because they are just matrices, but I think there are better names for W1/W2. Like seeds_A, seeds_B or something like that. But I definitely think should have the name "seeds" in it